src/EditorFeatures/CSharpTest/ImplementAbstractClass/ImplementAbstractClassTests.cs

```diff
@@ -1818,6 +1818,149 @@ public override void M<T>()
+    protected abstract T Method([System.Diagnostics.CodeAnalysis.AllowNullAttribute] T value);
+}
+
+class [|Class|] : Base<string>
+{
+}",
+@"#nullable enable
+
+abstract class Base<T>
+{
+    [return: System.Diagnostics.CodeAnalysis.MaybeNullAttribute]
+    protected abstract T Method([System.Diagnostics.CodeAnalysis.AllowNullAttribute] T value);
+}
+
+class Class : Base<string>
+{
+    protected override string Method(string? value)
+    {
+        throw new System.NotImplementedException();
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsImplementAbstractClass)]
+        public async Task TestAllowNullAttributeOnNullableReferenceType()
+        {
+            await TestInRegularAndScript1Async(
+@"#nullable enable
+
+abstract class Base<T>
+{
+    [return: System.Diagnostics.CodeAnalysis.MaybeNullAttribute]
+    protected abstract T Method([System.Diagnostics.CodeAnalysis.AllowNullAttribute] T value);
+}
+
+class [|Class|] : Base<string?>
+{
+}",
+@"#nullable enable
+
+abstract class Base<T>
+{
+    [return: System.Diagnostics.CodeAnalysis.MaybeNullAttribute]
+    protected abstract T Method([System.Diagnostics.CodeAnalysis.AllowNullAttribute] T value);
+}
+
+class Class : Base<string?>
+{
+    protected override string? Method(string? value)
+    {
+        throw new System.NotImplementedException();
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsImplementAbstractClass)]
+        public async Task TestFlowAnylasisAttributeOnNonNullableValueType()
+        {
+            await TestInRegularAndScript1Async(
+@"abstract class Base<T>
+{
+    [return: System.Diagnostics.CodeAnalysis.MaybeNullAttribute]
+    protected abstract T MethodAllowNull([System.Diagnostics.CodeAnalysis.AllowNullAttribute] T value);
+
+    [return: System.Diagnostics.CodeAnalysis.NotNullAttribute]
+    protected abstract T MethodDisallowNull([System.Diagnostics.CodeAnalysis.DisallowNullAttribute] T value);
+}
+
+class [|Class<T>|] : Base<T> where T : struct
+{
+}",
+@"abstract class Base<T>
+{
+    [return: System.Diagnostics.CodeAnalysis.MaybeNullAttribute]
+    protected abstract T MethodAllowNull([System.Diagnostics.CodeAnalysis.AllowNullAttribute] T value);
+
+    [return: System.Diagnostics.CodeAnalysis.NotNullAttribute]
+    protected abstract T MethodDisallowNull([System.Diagnostics.CodeAnalysis.DisallowNullAttribute] T value);
+}
+
+class Class<T> : Base<T> where T : struct
+{
+    protected override T MethodAllowNull(T value)
+    {
+        throw new System.NotImplementedException();
+    }
+
+    protected override T MethodDisallowNull(T value)
+    {
+        throw new System.NotImplementedException();
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsImplementAbstractClass)]
+        public async Task TestFlowAnylasisAttributeOnNullableValueType()
+        {
+            await TestInRegularAndScript1Async(
```

Comment:
Do we want a similar test where the base is unconstrained but the _derived_ type is constrained to class, like how this test is constraining to struct?

Reply:
Added with problematic results.